edge-util/src/Microsoft.Azure.Devices.Edge.Storage.RocksDb/disk/DiskSpaceChecker.cs

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Storage.RocksDb.Disk
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Storage.Disk;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class DiskSpaceChecker : IStorageSpaceChecker
+    {
+        static readonly int checkFrequencyDefault = 3;
+        readonly string storageFolder;
+        readonly object updateLock = new object();
+        Option<DiskSpaceCheckerBase> inner;
+        long maxSize;
+        Option<int> checkFrequency;
+
+        DiskSpaceChecker(string storageFolder, Option<DiskSpaceCheckerBase> inner)
+        {
+            this.storageFolder = Preconditions.CheckNonWhiteSpace(storageFolder, nameof(storageFolder));
+            this.inner = Preconditions.CheckNotNull(inner, nameof(inner));
+        }
+
+        public bool IsFull => this.inner.Match<bool>(b => b.DiskStatus == DiskSpaceStatus.Full, () => false);
+
+        public static DiskSpaceChecker Create(string storageFolder)
+        {
+            // Start up diskSpaceChecker unfilled - so that it will be enabled when SetMaxSizeBytes is called
+            var diskSpaceChecker = new DiskSpaceChecker(storageFolder, Option.None<DiskSpaceCheckerBase>());
+            Events.Created(storageFolder);
```

Comment:
This log might be misleading as the DiskSpaceChecker hasn't really been started. Consider moving this to the place where you actually 'start' the checker. You can rename it to Started or something if you'd like

Reply:
Changed it